lib/phoenix_live_view/tag_engine.ex

```diff
@@ -1215,10 +1217,44 @@ defmodule Phoenix.LiveView.TagEngine do
     List.keydelete(attrs, "phx-no-format", 0)
   end
 
+  defp validate_tag_attrs!(attrs, %{tag_name: "input"}, state) do
+    # warn if using name="id" on an input
+    case Enum.find(attrs, &match?({"name", {:string, "id", _}, _}, &1)) do
```

Comment:
yes, this means that we go through the attrs twice for input tags, but I didn't want to mix this into the phx attrs handling (and keep it simple)